flow/include/flow/serialize.h

```diff
@@ -831,14 +836,47 @@ struct PacketBuffer : SendBuffer {
 public:
 	static PacketBuffer* create(size_t size = 0) {
 		size = std::max(size, PACKET_BUFFER_MIN_SIZE - PACKET_BUFFER_OVERHEAD);
-		uint8_t* mem = new uint8_t[size + PACKET_BUFFER_OVERHEAD];
+		uint8_t* mem = nullptr;
+		if (keepalive_allocator::isActive()) [[unlikely]]
+			mem = static_cast<uint8_t*>(keepalive_allocator::allocate(size + PACKET_BUFFER_OVERHEAD));
+		else
+			mem = new uint8_t[size + PACKET_BUFFER_OVERHEAD];
 		return new (mem) PacketBuffer{ size };
 	}
+
 	PacketBuffer* nextPacketBuffer() { return static_cast<PacketBuffer*>(next); }
+
+	void markForWipe(uint8_t* begin, size_t size) {
+		ASSERT(data() <= begin);
+		if (!size)
+			return;
+		ASSERT(data() + this->size() >= begin + size);
+		uint32_t const this_wipe_begin = uint32_t(begin - data());
+		if (wipe_len == 0) {
+			wipe_begin = this_wipe_begin;
+			wipe_len = (uint32_t)size;
+		} else {
+			// Wipe range already exists. Take a contiguous union of the two ranges.
+			// This means any gaps between ranges will also be wiped
+			uint32_t const this_wipe_end = this_wipe_begin + (uint32_t)size;
+			uint32_t const old_wipe_end = wipe_begin + wipe_len;
+			uint32_t const new_wipe_begin = std::min(wipe_begin, this_wipe_begin);
+			uint32_t const new_wipe_end = std::max(old_wipe_end, this_wipe_end);
+			wipe_begin = new_wipe_begin;
+			wipe_len = new_wipe_end - new_wipe_begin;
+		}
+	}
+
 	void addref() { ++reference_count; }
 	void delref() {
 		if (!--reference_count) {
-			delete[] reinterpret_cast<uint8_t*>(this);
+			if (wipe_len > 0) {
+				::memset(data() + wipe_begin, 0, wipe_len);
+			}
+			if (keepalive_allocator::isActive()) [[unlikely]]
+				keepalive_allocator::invalidate(this);
```

Comment:
Consider hiding this conditional logic in some inline function declared in FastAlloc.h

Reply:
Good point. Will do.